Departments CRUD

gaschneider opened the pull request:
The changes in this PR consist in:

- Adding CRUD to departments
- Setting permissions for each department action
- Adding frontend to interact with api
- Display department management in frontend according to permissions
- Add test coverage to all actions of department controller

gaschneider committed:
add departments react and node, missing tests

gaschneider committed:
Add initial department tests (still missing specific permissions and user not authenticated)

gaschneider committed:
add new level in tests to easy read

gaschneider committed:
add department tests with permissions

magicblak commented:
Validation completed! The mentioned items are OK. It will only be necessary to uncomment the code responsible for synchronizing the database in the development or test environment, using the sequelize.sync method with the alter: true option. This code is located in the server's index.ts file. Please make the adjustments and proceed with the changes as needed.

gaschneider committed:
Address PR comment, update readme and generate initial migrations

gaschneider committed:
add command to apply migrations in the readme in case production environment is selected

gaschneider commented:
Addressed the commented code, update the readme file with the option to use `npm run dev` to run the server and also added the initial migrations for the db that were missing.

gaschneider commented:
> Validation completed! The mentioned items are OK. It will only be necessary to uncomment the code responsible for synchronizing the database in the development or test environment, using the sequelize.sync method with the alter: true option. This code is located in the server's index.ts file. Please make the adjustments and proceed with the changes as needed.

Addressed the commented code, update the readme file with the option to use npm run dev to run the server and also added the initial migrations for the db that were missing.
